app/controllers/admin/orders_controller.rb

```diff
@@ -0,0 +1,37 @@
+class Admin::OrdersController < Admin::BaseController
+  before_action :set_order, only: %i[deliver_order cancel_order]
+  
+  def index
+    if params[:email]
+      @orders = User.find_by(email: params[:email]).orders.not_in_rogress
+      flash[:notice] = 'No orders found' if @orders.empty?
+    else
+      @orders = Order.not_in_progress
+    end
+  end
+
+  def deliver_order
+    @order.delivered!
+    OrderMailer.with(order: @order).delivered.deliver_later
```

Comment:
Should these 2 lines be a method in `order.rb` too. 
WE can create a method `deliver` on order which takes a prama `send_email` as boolean